src/entities/friendship_history.rs

```diff
@@ -122,6 +123,57 @@ impl FriendshipHistoryRepository {
         }
     }
 
+    /// Query the history of `request` events for a friendship between two timestamps.
+    pub async fn get_friendship_request_event_history<'a>(
+        &'a self,
+        friendship_id: Uuid,
+        timestamp_from: NaiveDateTime,
+        timestamp_to: NaiveDateTime,
+        transaction: Option<Transaction<'a, Postgres>>,
+    ) -> (
+        Result<Vec<FriendshipHistory>, sqlx::Error>,
+        Option<Transaction<'a, Postgres>>,
+    ) {
+        let executor = self.get_executor(transaction);
+
+        // Buld query
+        let query = sqlx::query(
+            "SELECT * FROM friendship_history 
+                WHERE friendship_id = $1
+                  AND event = 'request' 
+                  AND metadata IS NOT NULL 
```

Comment:
What about if the metadata is empty because the user decided not to send a request message? Or are we handling this when the request event is created by adding an empty "{}"?

Reply:
Update: I ignored the part that you are using this for `#[get("/v1/friendships/{friendshipId}/request-events/messages")]`, but maybe you can add a flag in the parameters to get the ones with metadata null and the ones with metadata to reuse this function.

Reply:
Yes, good idea! I can already think of a scenario where I want to retrieve all `request` events independently of the metadata (to prevent a user sending the same friend request more than 'N' times).